dask_jobqueue/core.py

```diff
@@ -447,22 +453,47 @@ def __init__(
                 )
             )
 
+        if dashboard_address is not None:
```

Comment:
Not allowing to pass `host` and `dashboard_address` is a breaking change. I guess it feels cleaner to be able to pass all the scheduler options through `scheduler_options`. Also if we keep `host` for example, you can pass different things at different levels, e.g. `host` + `interface` in `scheduler_options` which would make the error hard to make user-friendly (and/or the code a bit ugly).

If you think that's too harsh, I can make it a warning until the 0.8 release!

Reply:
What feel's weird here is to keep the `host` and `dashboard_address` in the constructor kwargs, but thrown an exception when used. Maybe we should add some doc or comment, or simply remove them and catch them through `**kwargs` argument? Or else we should handle it and log a deprecation warning.

I'm not really familliar with Python best practices in this case.

Reply:
I can see two solutions:
1. raise an informative error if people are using `host` or `dasboard_address` to tell them to pass this through `scheduler_options`. What I had in mind was to have this user-friendly error in 0.7.1 and remove `host` and `dashboard_address` in 0.8.
2. warn if people are using `host` or `dashboard_address` telling them that this will not be supported in 0.8 and then combine them with `scheduler_options` to do the right thing.

I tend to think 1. is easier because you don't need the combining logic between `host` `dashboard_address` and `scheduler_options` which can be tricky. For example, if you use `host` and use `scheduler_options={'interface': eth0}` which one should have the priority and should we raise an error in this case.

Also my feeling is that people ignore warnings until it breaks so that 2. is not that helpful for users in the end.

> I'm not really familliar with Python best practices in this case.

It completely depends on the project. For example, in scikit-learn we have a strong backward-compatibility policy: we would have warnings for two major releases before we can break user code. I feel in `dask-jobqueue` we can afford to be less conservative.